src/main/java/uk/ac/ebi/embl/api/validation/submission/SubmissionValidationPlan.java

```diff
@@ -151,149 +160,185 @@ public void execute() throws ValidationEngineException {
-			return;
+			return result;
 		String fileName=null;
-
 		try
 		{
-			check = new FlatfileFileValidationCheck(options);
-			for(SubmissionFile flatfile:options.submissionFiles.get().getFiles(FileType.FLATFILE))
-			{
-				fileName= flatfile.getFile().getName();
-				if(sequenceDB!=null)
-					check.setSequenceDB(sequenceDB);
-				if(contigDB!=null)
-					check.setContigDB(contigDB);
-				if(!check.check(flatfile))
-					throwValidationCheckException(FileType.FLATFILE,flatfile);
-				else if(!options.isRemote)
-				     flagValidation(FileType.FLATFILE);
+		check = new FlatfileFileValidationCheck(options);
+		for(SubmissionFile flatfile:options.submissionFiles.get().getFiles(FileType.FLATFILE))
+		{
+			fileName= flatfile.getFile().getName();
+			if(sequenceDB!=null)
+				check.setSequenceDB(sequenceDB);
+			if(contigDB!=null)
+				check.setContigDB(contigDB);
+			result = check.check(flatfile);
+			if(!result.isValid()) {
+				result.setDefaultOrigin(new DefaultOrigin(getValidationErrorMessage(FileType.FLATFILE,flatfile, check)));
+				return result;
 			}
+		}
+			if (!options.isRemote)
+				flagValidation(FileType.FLATFILE);
+		}catch(ValidationEngineException e){
+			throwValidationEngineException(FileType.FLATFILE,e,fileName);
 		}catch(Exception e)
 		{
-			throwValidationEngineException(FileType.FLATFILE,e,fileName);
+			throw new ValidationEngineException();
 		}
+		return result;
 	}
 
-	private void validateAGP() throws ValidationEngineException
+	private ValidationResult validateAGP() throws ValidationEngineException
 	{
+		ValidationResult result = new ValidationResult();
 		if(options.processDir.isPresent()&&Files.exists(Paths.get(String.format("%s%s%s",options.processDir.get(),File.separator,agpFlagFileName))))
-			return;
+			return result;
 		String fileName=null;
 		try
 		{
-			for(SubmissionFile agpFile:options.submissionFiles.get().getFiles(FileType.AGP))
-			{
-				fileName= agpFile.getFile().getName();
-				if(sequenceDB!=null)
-					agpCheck.setSequenceDB(sequenceDB);
-				if(!agpCheck.check(agpFile))
-					throwValidationCheckException(FileType.AGP,agpFile);
-				else if(!options.isRemote)
-				     flagValidation(FileType.AGP);
-			}
-		}catch(Exception e)
+		for(SubmissionFile agpFile:options.submissionFiles.get().getFiles(FileType.AGP))
 		{
+			fileName= agpFile.getFile().getName();
+			if(sequenceDB!=null)
+				agpCheck.setSequenceDB(sequenceDB);
+			result = agpCheck.check(agpFile);
+			if(!result.isValid()) {
+				result.setDefaultOrigin(new DefaultOrigin(getValidationErrorMessage(FileType.AGP,agpFile, agpCheck)));
+				return result;
+			}
+		}
+			if (!options.isRemote)
+				flagValidation(FileType.AGP);
+
+		} catch (ValidationEngineException e) {
 			throwValidationEngineException(FileType.AGP,e,fileName);
 		}
+		catch(IOException e)
+		{
+			throw new ValidationEngineException(e);
+		}
```

Comment:
Now sure why we don't just do:

catch(Exception e)
{
throwValidationEngineException(FileType. AGP,e,"master.dat");
}